apps/src/lib/node/ledger/shell/process_proposal.rs

```diff
@@ -21,32 +23,65 @@ where
         Default::default()
     }
 
-    /// Check all the txs in a block. Some txs may be incorrect,
-    /// but we only reject the entire block if the order of the
-    /// included txs violates the order decided upon in the previous
-    /// block.
+    /// Check all the txs in a block.
+    /// We reject the entire block when:
+    ///    - decrypted txs violate the committed order
+    ///    - more decrypted txs than expected
+    ///    - checks on wrapper tx fail
+    ///
+    /// We cannot reject the block for failed checks on the decrypted txs since
+    /// their order has already been committed in storage, so we simply discard
+    /// the single invalid inner tx
     pub fn process_proposal(
         &self,
         req: RequestProcessProposal,
     ) -> ProcessProposal {
-        let tx_results = self.process_txs(&req.txs);
+        let tx_results =
+            self.process_txs(&req.txs, self.get_block_timestamp(req.time));
 
         ProcessProposal {
-            status: if tx_results.iter().any(|res| res.code > 3) {
-                ProposalStatus::Reject as i32
-            } else {
+            status: if tx_results.iter().all(|res| {
```

Comment:
Ah here. If the ordering of the errors is fixed, we just need to change the `3` to a `5` in the old code

Reply:
I think I'm missing something, what do you mean when you say "in the old code"?

Reply:
```            status: if tx_results.iter().any(|res| res.code > 3) {
                ProposalStatus::Reject as i32
            } else {
            ```

Reply:
Ah, I think I see what you mean. I've changed that code to use the enum variants because I was struggling to keep track of the error codes and thought that using the variants was more readable. Would you prefer to revert to using the codes instead?

Reply:
We rely on it heaving in `eth-bridge-integration`, so I thinks it's easier for you to make this change than us.

Reply:
Actually, in this PR I've also changed `process_proposal` to reject the block in more cases: in particular to reject a block even in case of an `InvalidTx` or `InvalidSig` which we were not doing before. Does this look ok to you? In theory, the block proposer should be able to verify these at block construction time. So in essence the threshold could stay at 3 like before if I reorder the variants correctly in the enum

Reply:
Yep, sounds good